specs/windows/ads.table

```diff
@@ -0,0 +1,13 @@
+table_name("ads")
+description("Returns the stream names and values for files using NTFS Alternate Data Streams (ADS).")
+schema([
+    Column("path", TEXT, "Absolute file path", required=True, index=True),
+    Column("directory", TEXT, "Directory of file(s)", required=True),
+    Column("key", TEXT, "Name of the value generated from the stream"),
+    Column("value", TEXT, "The parsed information from the attribute"),
```

Comment:
What kind of data shows up here? Some internet things talk about malware smuggling entire file contents here. Will that be okay to push back in a column? (We don't generally push that much data through osquery, so it feels a little amiss)

Reply:
100% agree. I think the main value of this table is the content of the `Zone.Identifier` stream which can help during investigations to identify where the file was downloaded from.
In my PR description I left an open question about how to handle cases where a stream contains an entire file. Maybe we can set a hard limit on the length of the content and warn users if the content is too large to be displayed by osquery? 
Happy to hear other thoughts.

I'm also not sure how the `extended_attributes` table handles this type of cases in nix systems.

Reply:
> I'm also not sure how the `extended_attributes` table handles this type of cases in nix systems.

The possible value size is much more limited: https://en.wikipedia.org/wiki/Extended_file_attributes

>The Linux kernel allows extended attribute to have names of up to 255 bytes and values of up to 64 KiB, as do XFS and ReiserFS, but ext2/3/4 and btrfs impose much smaller limits, requiring all the attributes (names and values) of one file to fit in one "filesystem block" (usually 4 KiB)

Reply:
I think it probably depends on what we're concerned about. 

If it's _content_, the most conservative approach would be to only fetch the `value` for keys on a compiled allowlist. (pushing people to use carves if they want the content)

If it's size, truncation and warnings probably make sense.